lib/dateformat.js

```diff
@@ -0,0 +1,255 @@
+/*
```

Comment:
This change is introducing a new file `dateformat.js` under the `lib` directory.  However, the project is built around using the `dateformat.js` file in the `src` directory (see the main source code for reference).  Instead of creating a new file, can we please change this PR to make changes to the exiting `src/dateformat.js` file 😄